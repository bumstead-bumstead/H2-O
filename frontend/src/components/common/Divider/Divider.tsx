import type { CSSProperties, HTMLAttributes } from 'react';
import styled from '@emotion/styled';
import { Colors } from '@/styles/colors';

interface Props extends HTMLAttributes<HTMLDivElement> {
  vertical?: boolean;
  length: CSSProperties['width'];
  color?: Colors;
}

const Divider = styled.div<Props>(({ theme, length, vertical, color }) => ({
  backgroundColor: color ? theme.colors[color] : theme.colors.gray400,
  width: vertical ? '1px' : length,
  height: vertical ? length : '1px',
}));

export default Divider;
