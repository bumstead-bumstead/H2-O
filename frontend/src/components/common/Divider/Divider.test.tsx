import Divider from './Divider';
import { render, screen } from '@/tests/test-util';

describe('Divider 컴포넌트 테스트', () => {
  it('Divider 정상적으로 렌더링된다.', () => {
    render(<Divider data-testid='divider' length='100%' />);

    expect(screen.getByTestId('divider')).toBeInTheDocument();
  });

  describe('Divider 스냅샷 테스트', () => {
    it('horizontal 일 때', () => {
      render(<Divider data-testid='divider' length='100%' />);

      expect(screen.getByTestId('divider')).toMatchSnapshot();
    });

    it('vertical 일 때', () => {
      render(<Divider data-testid='divider' length='100%' vertical />);

      expect(screen.getByTestId('divider')).toMatchSnapshot();
    });

    it('color 가 주어졌을 때', () => {
      render(<Divider data-testid='divider' length='100%' color='activeBlue' />);

      expect(screen.getByTestId('divider')).toMatchSnapshot();
    });
  });
});
