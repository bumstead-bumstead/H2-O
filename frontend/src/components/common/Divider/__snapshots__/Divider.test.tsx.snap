// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider 컴포넌트 테스트 > Divider 스냅샷 테스트 > color 가 주어졌을 때 1`] = `
.emotion-0 {
  background-color: #00AAD2;
  width: 100%;
  height: 1px;
}

<div
  class="emotion-0"
  color="activeBlue"
  data-testid="divider"
/>
`;

exports[`Divider 컴포넌트 테스트 > Divider 스냅샷 테스트 > horizontal 일 때 1`] = `
.emotion-0 {
  background-color: #B6B8BB;
  width: 100%;
  height: 1px;
}

<div
  class="emotion-0"
  data-testid="divider"
/>
`;

exports[`Divider 컴포넌트 테스트 > Divider 스냅샷 테스트 > vertical 일 때 1`] = `
.emotion-0 {
  background-color: #B6B8BB;
  width: 1px;
  height: 100%;
}

<div
  class="emotion-0"
  data-testid="divider"
/>
`;
